packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -744,16 +708,89 @@ void _handleXCResultIssue({required XCResultIssue issue, required Logger logger}
       break;
   }
 
-  // Add more custom output for flutter users.
-  if (issue.message != null && issue.message!.toLowerCase().contains('provisioning profile')) {
+  if (issue.message == null) {
+    return _XCResultIssueHandlingResult(requiresProvisioningProfile: false, hasProvisioningProfileIssue: false);
+  }
+
+  // Add more error messages for flutter users for some special errors.
+  final String message = issue.message!;
```

Comment:
If you make the local before the null check you won't need the `!`
```suggestion
  // Add more error messages for flutter users for some special errors.
  final String? message = issue.message;
  
  if (issue.message == null) {
    return _XCResultIssueHandlingResult(requiresProvisioningProfile: false, hasProvisioningProfileIssue: false);
  }
```